lib/buffer.dart

```diff
@@ -140,6 +140,12 @@ class BytesBuffer {
     add([byte]);
   }
 
+  /// Add a byte array to buffer at [index]
+  void insert(int index, List<int> bytes, {bool copy}) {
+    _chunks.insert(index, castBytes(bytes, copy: copy ?? _copy));
```

Comment:
I believe this does not do what the intended use does: `_chunks` are internal to the builder, and an index in the chunks is not the same as a positional index for the overall buffer.

I think if you write a few tests for it, it would be clear how this implementation won't work the way you intend it to. The insert seems to be a costly operation, it requires two or more array copy to get it working.